packages/@aws-cdk/aws-appmesh/test/test.virtual-gateway.ts

```diff
@@ -153,6 +155,151 @@ export = {
       }));
       test.done();
     },
+    'with an http listener with a TLS certificate from ACM'(test: Test) {
+      // GIVEN
+      const stack = new cdk.Stack();
+
+      // WHEN
+      const mesh = new appmesh.Mesh(stack, 'mesh', {
+        meshName: 'test-mesh',
+      });
+
+      const cert = new Certificate(stack, 'cert', {
+        domainName: '',
+      });
+
+      new appmesh.VirtualGateway(stack, 'testGateway', {
+        virtualGatewayName: 'test-gateway',
+        mesh: mesh,
+        listeners: [appmesh.VirtualGatewayListener.httpGatewayListener({
+          port: 8080,
+          tlsCertificate: appmesh.TlsCertificate.acm({
+            tlsMode: TlsMode.STRICT,
+            acmCertificate: cert,
+          }),
+        })],
+      });
+
+      // THEN
+      expect(stack).to(haveResourceLike('AWS::AppMesh::VirtualGateway', {
+        Spec: {
+          Listeners: [
+            {
+              PortMapping: {
+                Port: 8080,
+                Protocol: appmesh.Protocol.HTTP,
+              },
+              TLS: {
+                Mode: 'STRICT',
```

Comment:
I think we can use enum here too.

Reply:
hm ok I'll check at how Protocol is done